src/lib.rs

```diff
@@ -128,7 +129,6 @@ pub fn desugar_book(
   ctx.check_unbound_vars()?;
 
   ctx.book.make_var_names_unique();
-  ctx.book.desugar_use();
```

Comment:
Doesn't linearize matches insert some `use` terms?
We still need to call `desugar_use` after it, no?